### PR TITLE
Update to prevent error for base64 encoded signature passed in via arg

### PR DESCRIPTION
**Pull request by font**
I was getting a `no such file or directory` error when passing in the raw base64 encoded signature via command line arg as shown https://github.com/sigstore/cosign#upload-a-generated-signature. It's because the error signals to interpret the argument as a raw base64 encoded signature instead of a file or `stdin`. So this clears the error to avoid inadvertently returning an error and exiting the program.

### CODE REVIEW

File: cmd/cli/upload.go [hunk 1]
```diff
@@ -61,6 +61,7 @@ func UploadCmd(ctx context.Context, sigRef, payloadRef, imageRef string) error {
 		b64SigBytes, err = ioutil.ReadAll(os.Stdin)
 	} else if _, err = os.Stat(sigRef); os.IsNotExist(err) {
 		b64SigBytes = []byte(sigRef)
+		err = nil
```
Comment (jonjohnsonjr) [inline cmd/cli/upload.go:64]: I think this is too tricky. I'd rather see this code cleaned up a bit to make it clearer what's going on.

Comment (dlorenc) [reply cmd/cli/upload.go:64]: oof yea I apologize for this code.

**Comment by dlorenc** [comment]
@font - let me know if you want to pair for a bit tomorrow on cleaning this up. I'd also appreciate talking it through to make sure I even understand what I was trying to do :)

File: cmd/cli/upload.go (hunk 1 above)
Comment (font) [reply cmd/cli/upload.go]: No worries!

**Comment by font** [comment]
@jonjohnsonjr @dlorenc I've pushed an update that attempts to clean this up and added test cases for it. Let me know if you had something else in mind. Thanks @dlorenc - happy to discuss further tomorrow.

**Review by dlorenc — APPROVED** [review]
Oh wow, much nicer!

**Pull request merged**